src/IncompleteTasksOnly.js

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import {ChecklistItem} from "./ChecklistItem";
+
+export function IncompleteTasksOnly(props) {
+    const initialCompletedItems = props.items.map((item) => item.completed ? item.id : undefined);
```

Comment:
initialCompletedItems should always be  correct as long as you've updated the checked state of the data.  No need for completedItems maintenance.